### DISCUSSION TITLE
How to boot Ubuntu 22.04 from NVME drive installed on OrangePi 5 Pro (no SPI flash) and an SD Card

### DISCUSSION THREAD

**Discussion by Tekunalogy** [Joshua-Riek/ubuntu-rockchip#821] [General]
1. Install NVME drive on OrangePi 5 Pro
2. Flash image to SD Card: https://joshua-riek.github.io/ubuntu-rockchip-download/
    - I used `ubuntu-22.04-preinstalled-server-arm64-orangepi-5-pro.img.xz`, since I am only using this via SSH
3. Install SD card into OrangePi 5 Pro
4. Log into and update
    - Login Info: https://github.com/Joshua-Riek/ubuntu-rockchip?tab=readme-ov-file#login-information
    - run `sudo apt update && sudo apt upgrade` (this is to ensure that the next command works)
5. Run this command to copy the OS onto the NVME drive: https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/Ubuntu-22.04-LTS#install-ubuntu-onto-an-nvme-from-linux
6. After it is complete, shutdown OrangePi 5 Pro
7. Remove SD card
8. Attach to a PC and use a disk partition management tool
9. Delete only the partition that contains "-rootfs". Do not delete any other partition.
    - Example: "desktop-rootfs"
    - In my case, that partition was also the largest partition.
10. Eject SD card, and install into OrangePi 5 Pro.
11. It will now boot into the OS on the NVME drive

**Comment by watou** [comment]
Is the device named RockPi5 Pro?

**Reply by Tekunalogy** [reply]
Apologies, I meant to write OrangePi 5 Pro, not rockpi. Updated the title

**Comment by Jesterovskiy** [comment]
```
/dev/disk4 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme            *31.3 GB    disk4
   1:                        EFI                                 7.2 GB     disk4s1
                    (free space)                         24.0 GB    -
```
This is what I see on SD, and I don't understand how to properly delete the partition)

**Reply by Tekunalogy** [reply]
My main Desktop PC also runs ubuntu 22.04, so I deleted the partition on the sd card using the "Disks" application on ubuntu.

**Reply by Jesterovskiy** [reply]
Thanks) Done with diskutil on macos:
```
diskutil list
...
/dev/disk5 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *31.3 GB    disk5
   1:                        EFI                         31.2 GB    disk5s1
```

```
diskutil eraseVolume free free disk5s1
```

**Comment by Tekunalogy** [comment]
Sorry, I meant to say OrangePi 5 Pro, not rockpi. Updated the title

**Comment by furyhawk** [comment]
look like there's no way to boot from emmc to nvme `/`.

**Reply by furyhawk** [reply]
found the way to boot from emmc to nvme /.

between step 4 and 5. 
- repeat the process(copy linux) for emmc.
- shutdown and boot from sd card.
- replace step 8, `delete only the partition that contains "-rootfs" ` from the emmc. It display the emmc drive as `SD Card Reader`.
- Reboot without the SD card.

**Comment by fiko** [comment]
boom... it works 🥳 👍 💯

**Comment by IgriegaL** [comment]
its works for me in august 2025 thanks!

**Comment by sunnychase** [comment]
I am trying to boot from the usbc and leave the pcie slot for another board application.

**Comment by akmaldira** [comment]
on step 9 (i use macos)

 
```sh
diskutil list
// internal disk
/dev/disk4 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *31.1 GB    disk4
   1:       Microsoft Basic Data CIDATA                  4.2 MB     disk4s1
   2:                        EFI                         31.1 GB    disk4s2
```

i don't see any `*-rootfs`, so i'm confuse which one should i delete
